### PR TITLE
Significantly speed up create & destroy by reducing the amount of time to wait for GC

### PR DESCRIPTION
**Pull request by Mothblocks**
The check queue is 5 minutes long because that's the longest a client can hold onto a reference. Without clients, we can drastically decrease the time we have to wait. This lowers the time down to 10 seconds (though everything right now deletes in 5). 

This will represent a 5 minute decrease in CI across the board, freeing up runners.

Makes a few changes to stuff that was being held for more than 10 seconds.
- `VARSET_CALLBACK` now works through weakrefs, to allow for pAIs to have their holochassis init timers.
- Nar'Sie cleans herself up in GLOB.cult_narsie if she's deleted.
- "Spooky portals" no longer hold onto a reference for 2 minutes.
- `poll_candidates` short circuits to an empty list if there are no candidates, to avoid several 30 second+ long timers

Originally this was going to be a more clever hack from @MrStonedOne about short circuiting if everything deletes before the wait, but we realized that basically nothing actually holds onto references for that long without clients, and that nothing really should anyway

### CODE REVIEW

**Comment by Mothblocks** [comment]
Turns out this works on 514 too

**Comment by Mothblocks** [comment]
image

**Comment by Mothblocks** [comment]
Tha'ts not 515 that's 514 Runtime--515 is even shorter at around 6 minutes flat

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -238,6 +238,9 @@
 
 ///Calls the show_candidate_poll_window() to all eligible ghosts
 /proc/poll_candidates(question, jobban_type, be_special_flag = 0, poll_time = 300, ignore_category = null, flashwindow = TRUE, list/group = null)
+	if (group.len == 0)
```
Comment (ZephyrTFA) [inline code/__HELPERS/game.dm:241]:
```suggestion
	if (group?.len == 0)
```
The args show that this can be null

Comment (Mothblocks) [reply code/__HELPERS/game.dm:241]: `null` makes absolutely no sense as a value here and I would rather it runtime. I don't think the args show anything other than that `group` was added later on, and you can't have parameters after ones with a default. Notice how every caller uses `group = `, or passes them all.

Comment (ZephyrTFA) [reply code/__HELPERS/game.dm:241]: Sounds good to me

**Pull request merged**